quit/web/modules/endpoint.py

```diff
@@ -119,44 +158,85 @@ def provenance():
     if 'Accept' in request.headers:
         mimetype = parse_accept_header(request.headers['Accept']).best
     else:
-        mimetype = 'text/html'
+        mimetype = '*/*'
 
     if q:
         try:
             query_type = parse_query_type(q)
             graph = quit.store.store
 
-            if query_type in ['SELECT', 'CONSTRUCT', 'ASK', 'DESCRIBE']:
+            if query_type in ['SELECT', 'ASK']:
                 res = graph.query(q)
 
-                if mimetype in ['text/html', 'application/xhtml_xml', '*/*']:
+                # Set default
+                if mimetype == '*/*':
+                    mimetype = 'application/sparql-results+xml'
+
+                if mimetype in ['text/html', 'application/xhtml_xml']:
                     results = res.serialize(format='html')
                     response = make_response(render_template(
                         "results.html", results=Markup(results.decode())))
                     response.headers['Content-Type'] = 'text/html'
                     return response
-                elif mimetype in ['application/json', 'application/sparql-results+json']:
+                elif mimetype in ['application/sparql-results+json']:
```

Comment:
https://www.w3.org/TR/sparql11-results-json/

Please support `application/json` and `application/sparql-results+json`